pkg/filelock/filelock.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filelock
+
+import (
+	"os"
+	"syscall"
+
+	"github.com/pingcap/errors"
+	cerrors "github.com/pingcap/ticdc/pkg/errors"
+)
+
+// FileLock represents a file lock created by `flock`.
+// For more on the system call `flock`, read:
+// https://linux.die.net/man/2/flock
+type FileLock struct {
+	fd *os.File
+}
+
+// NewFileLock creates a new file lock on the file described in filePath.
+func NewFileLock(filePath string) (*FileLock, error) {
+	file, err := os.OpenFile(filePath, os.O_RDWR|os.O_CREATE|syscall.O_NONBLOCK, 0o666)
```

Comment:
Please keep minimal permission requirements.
```suggestion
	file, err := os.OpenFile(filePath, os.O_RDWR|os.O_CREATE|syscall.O_NONBLOCK, 0o600)
```